src/MySqlConnector/MySqlClient/MySqlCommand.cs

```diff
@@ -111,7 +114,70 @@ protected override DbParameter CreateDbParameter()
 
 		internal async Task<int> ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken)
 		{
-			using (var reader = (MySqlDataReader) await ExecuteReaderAsync(CommandBehavior.Default, ioBehavior, cancellationToken).ConfigureAwait(false))
+			if (CommandType == CommandType.StoredProcedure)
+				return await ExecuteStoredProcedureAsync(ioBehavior, cancellationToken);
+
+			return await ExecuteTextCommandNonQueryAsync(CommandText, m_parameterCollection, ioBehavior, cancellationToken);
```

Comment:
Thanks, I'll definitely do a global grep for this before I push the final changeset